sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/ResourceConfiguration.java

```diff
@@ -11,25 +11,69 @@
 import io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException;
 import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
 import io.opentelemetry.sdk.autoconfigure.spi.internal.ConditionalResourceProvider;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.DefaultConfigProperties;
 import io.opentelemetry.sdk.resources.Resource;
 import io.opentelemetry.sdk.resources.ResourceBuilder;
 import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
 import java.io.UnsupportedEncodingException;
 import java.net.URLDecoder;
 import java.nio.charset.StandardCharsets;
+import java.util.Collections;
 import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
 import java.util.function.BiFunction;
 
 /** Auto-configuration for the OpenTelemetry {@link Resource}. */
-final class ResourceConfiguration {
+public final class ResourceConfiguration {
```

Comment:
Is your thinking that if we have any other resources that we want to auto-configured in this module, that we could add more methods? Otherwise, it might make sense to name this "EnvironmentResource"?

Reply:
My thinking is that this module should be reserved for interpreting the [spec environment variables](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md) which do not have a natural home in a different module. For example, the zipkin, otlp, jaeger, env variables belong in those respective modules.

Users should be able to access components configured according to the environment with or without `AutoConfiguredOpenTelemetrySdk`. This means they can cherry pick their config - maybe they do programatic config everywhere except the environment resource (as you've done), or except span limits. 

My first pass was to add an `EnvironmentResource` class, but on further consideration, that pattern doesn't scale when we get requests to access other components configured from the environment, like SpanLimits. It'd be more natural to add a public method `TracerProviderConfiguration#createSpanLimits(ConfigProperties)` than to add a dedicated `EnvironmentSpanLimits` class. 

I don't anticipate any other types of resource configuration being done from environment variables, so I'd be surprised if we ever had to add more methods.

Reply:
cool. I don't know that I have a much better name at this point, so let's move forward with this and hopefully get some feedback from users!